libs/gl-plugin/src/tramp.rs

```diff
@@ -347,6 +339,45 @@ pub async fn trampolinepay(
     }
 }
 
+fn pick_channels(
+    amount_msat: u64,
+    mut channels: Vec<ChannelData>,
+) -> Result<Vec<(cln_rpc::primitives::ShortChannelId, u64)>> {
+    let mut acc = 0;
+    let mut choosen = vec![];
+    while let Some(channel) = channels.pop() {
+        if acc == amount_msat {
+            break;
+        }
+
+        // Filter out channels that lack minimum funds and can not send an htlc.
+        if std::cmp::max(MIN_HTLC_AMOUNT, channel.min_htlc_out_msat) > channel.spendable_msat {
+            debug!("Skip channel {}: has spendable_msat={} and minimum_htlc_out_msat={} and can not send htlc.",
+                channel.short_channel_id,
+                channel.spendable_msat,
+                channel.min_htlc_out_msat,
+            );
+            continue;
+        }
+
+        if (channel.spendable_msat + acc) <= amount_msat {
+            choosen.push((channel.short_channel_id, channel.spendable_msat));
+            acc += channel.spendable_msat;
+        } else {
+            let rest = amount_msat - acc;
+            choosen.push((channel.short_channel_id, rest));
+            acc += rest;
```

Comment:
There's still a case where `rest` does not add up to the minimum htlc amount. Do you know a good way to handle that?

Reply:
Right, we may want to check and skip those channels as well. :thinking: 
I'll add a follow up.